allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/project_cloud_api_base.py

```diff
@@ -0,0 +1,190 @@
+from typing import Union, Optional
+from pathlib import Path
+import logging
+
+from allensdk.api.cloud_cache.cloud_cache import (
+    S3CloudCache, LocalCache, StaticLocalCache)
+
+from allensdk.brain_observatory.behavior.behavior_project_cache \
+    .utils import version_check
+
+
+class ProjectCloudApiBase(object):
+    """API for downloading data released on S3 and returning tables.
+
+    Parameters
+    ----------
+    cache: S3CloudCache
+        an instantiated S3CloudCache object, which has already run
+        `self.load_manifest()` which populates the columns:
+          - metadata_file_names
+          - file_id_column
+    skip_version_check: bool
+        whether to skip the version checking of pipeline SDK version
+        vs. running SDK version, which may raise Exceptions. (default=False)
+    local: bool
+        Whether to operate in local mode, where no data will be downloaded
+        and instead will be loaded from local
+    """
+    def __init__(
+        self,
+        cache: Union[S3CloudCache, LocalCache, StaticLocalCache],
+        skip_version_check: bool = False,
+        local: bool = False
+    ):
+
+        self.cache = cache
+        self.skip_version_check = skip_version_check
+        self._local = local
+        self.load_manifest()
+
+    def load_manifest(self, manifest_name: Optional[str] = None):
+        """
+        Load the specified manifest file into the CloudCache
+
+        Parameters
+        ----------
+        manifest_name: Optional[str]
+            Name of manifest file to load. If None, load latest
+            (default: None)
+        """
+        if manifest_name is None:
+            self.cache.load_last_manifest()
+        else:
+            self.cache.load_manifest(manifest_name)
+
+        if self.cache._manifest.metadata_file_names is None:
+            raise RuntimeError("S3CloudCache object has no metadata "
+                               "file names. BehaviorProjectCloudApi "
```

Comment:
Try replacing `BehaviorProjectCloudApi` with `{type(self)}` so that the actual class triggering this error gets written out. Should avoid confusion.

Reply:
Also: given that you run `load_manifest()` in lines 51-54, the last line of this error message should be changed. The problem isn't that `load_manifest()` has not been run. There just aren't any metadata files. The reason is likely unknown.

Reply:
I updated the error message